__tests__/app.test.js

```diff
@@ -3,93 +3,240 @@ const app = require("../db/app.js");
+    await request(app).get("/api/3/comments").expect(200);
+  });
+
+  test("comments sorted by most recent first", async () => {
+    const response = await request(app).get("/api/3/comments").expect(200);
+    const { comments } = response.body;
+    expect(comments).toBeSortedBy("created_at", {
+      descending: true,
+    });
+  });
+
+  test("returns array of comment objects for article id", async () => {
+    const response = await request(app).get("/api/3/comments").expect(200);
+    const { comments } = response.body;
+    expect(comments.length).toBe(2);
+    expect(comments).toMatchObject([
+      {
+        body: "Ambidextrous marsupial",
+        votes: 0,
+        author: "icellusedkars",
+        article_id: 3,
+        created_at: "2020-09-19T23:10:00.000Z",
+      },
+      {
+        body: "git push origin master",
+        votes: 0,
+        author: "icellusedkars",
+        article_id: 3,
+        created_at: "2020-06-20T07:24:00.000Z",
+      },
+    ]);
+  });
+});
+describe("POST /api/articles/:article_id/comments", () => {
+  test("returns 400 error message for invalid article id data type", async () => {
+    const postData = {
+      username: "butter_bridge",
+      body: "hello",
+    };
+   const response = await request(app)
+      .post("/api/articles/nn/comments")
+      .send(postData)
+      .expect(400);
+  expect(response.body.msg).toBe("400 Bad Request");
+  });
+  test("returns 404 error message for non-existent article id", async () => {
+    const postData = {
+      username: "butter_bridge",
+      body: "hello",
+    };
+   const response = await request(app)
+      .post("/api/articles/18/comments")
+      .send(postData)
+      .expect(404);
+  expect(response.body.msg).toBe("404 article_id of 18 Not Found");
+  });
+  test("returns 404 error for non-existent user", async () => {
+    const postData = {
+      username: "bob",
+      body: "hello",
+    };
+    const response = await request(app)
+      .post("/api/articles/6/comments")
+      .send(postData)
+      .expect(404);
+    expect(response.body.msg).toBe("404 username of bob Not Found");
+  });
+  test("returns 400 error for invalid request body", async () => {
+    const postData = {};
+    const response = await request(app)
+      .post("/api/articles/6/comments")
+      .send(postData)
+      .expect(400);
+    expect(response.body.msg).toBe("400 Bad Request");
+  });
+  test("returns comment object", async () => {
+    const postData = {
+      username: "butter_bridge",
+      body: "hello",
+    };
+    const response = await request(app)
+      .post("/api/articles/6/comments")
+      .send(postData)
+      .expect(201);
+    expect(response.body.comment).toBe("hello");
+  });
+  test("updates database", async () => {
+    const original = await checkExists("comments", "article_id", 6);
+    const postData = {
+      username: "butter_bridge",
+      body: "hello5",
+    };
+    await request(app)
```

Comment:
here you should return the posted comment from your controller so you should be able to check it using const postedCommet = await request(app).post...
and access it that way